src/Security/Security.php

```diff
@@ -438,6 +438,14 @@ public static function permissionFailure($controller = null, $messageSet = null)
     }
 
     /**
+     * The intended uses of this function is to temporarily change the current user for things such as
+     * canView() checks or unit tests.  It is stateless and will not persist between requests.  Importantly
+     * it also will not call any logic that may be present in the current IdentityStore logIn() or logout() methods
+     *
+     * If you are unit testing and calling FunctionalTest::get() or FunctionalTest::post() and you need to change
+     * the current user, you should instead use SapphireTest::logInAs() / logOut() which itself will call
+     * Injector::inst()->get(IdentityStore::class)->logIn($member) / logout()
+     *
```

Comment:
I think a method's PHP doc should first and foremost describe what the method does. Most of this content, while useful would make more sense in the [functional testing documentation](https://docs.silverstripe.org/en/4/developer_guides/testing/functional_testing/#loginas) page. 

```suggestion
     * Define the current user for the rest of this request. This method is stateless and will not persist between requests.
```

Reply:
That's probably correct, though I think people are likely to miss this information if it's in the functional testing docs rather than directly on this method.  There's are many places where Security::setCurrentUser() is called, and it this new failure with session manager installed is hard to diagnose if you're not familiar with the issue.  I think it's better to leave this verbose explanation in place because it's more likely to help people

Reply:
I've put an additional update into the functional testing docs to mention not to use the other methods that wont run login authenticator logic